pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -465,6 +465,12 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 		dexCfg = render.NewDexRelyingPartyConfig(authentication, dexTLSSecret, dexSecret, r.clusterDomain)
 	}
 
+	var oidcUserConfigMapAvailable, oidcUserSecretAvailable bool
+	if managementClusterConnection == nil {
+		oidcUserConfigMapAvailable = r.oidcUserConfigMapAvailable(ctx)
```

Comment:
Could you please follow the pattern that the other components use and pass the component to render and let it decide what to do if the component exists or not.

For instance, the config map might exist but in future iterations there may be upgrades that we have to make to the config map or changes to make, like adding annotations or labels or completely restructuring the resource.

The general pattern has been to grab a resource, if it exists re render it anyway because you may have updates to it or somebody may have changed it outside of the operator and we want to reconcile it. With the config map and secret we do want the Data edited, but the operator can make sure that config map a secret meta data that should not be changed is always accurate.

Also to note, if the data is changed while the operator is reconciling the request to update the config map will fail and be re attempted later because the revision version will be different, so this is a safe pattern.